include/sdf/Element.hh

```diff
@@ -559,11 +559,6 @@ namespace sdf
     /// \param[in] _child Pointer to the child to remove.
     public: void RemoveChild(ElementPtr _child);
 
-    /// \brief Remove a child element.
-    /// \param[in] _child Pointer to the child to remove.
-    /// \param[out] _errors Vector of errors.
-    public: void RemoveChild(ElementPtr _child, sdf::Errors &_errors);
```

Comment:
We can't remove this because it breaks ABI :(.

Reply:
Oh true! shall we deprecate it instead and remove it on the the next release?

Reply:
I'll open a PR to `main` then. Would an `sdf::Error` with an `ErrorCode::WARNING` message suffice?

Reply:
Not sure what the `ErrorCode::WARNING` would be for.

Reply:
Deprecation message?

Reply:
Oh, we'd use the `SDF_DEPRECATED` macro for that.

Reply:
`GZ_DEPRECATED` maybe?

Reply:
Yeah, `GZ_DEPRECATED`. I guess `SDF_DEPRECATED` is deprecated 😁 

We can discuss this on the followup PR, but if there's a chance the `sdf::Error` might be used by `RemoveChild`, it might be better to deprecate the overload that doesn't take `sdf::Error`.